react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSumbit = this.handleSumbit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({
+      email: event.target.value
+    });
+  }
+
+  handleSumbit(event) {
+    console.log('state:', this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSumbit}>
+        <label>Email</label>
+        <input
+        type='email'
+        onChange={this.handleChange}>
```

Comment:
be sure to also pass a `value` prop of `this.state.email` so that the component is fully "controlled".